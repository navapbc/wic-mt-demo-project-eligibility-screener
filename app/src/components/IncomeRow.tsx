import { ReactElement } from 'react'

export type IncomeRowProps = {
  periods: string[]
  householdSize: string
  incomeForHouseholdSize: {
    [key: string]: string
  }
}

const IncomeRow = (props: IncomeRowProps): ReactElement => {
  const { periods, householdSize, incomeForHouseholdSize } = props

  // If the householdSize is not an empty string, then use it
  // to lookup the income amounts for each time period.
  // Otherwise, return a placeholder row.

  return (
    <tr>
      {periods.map((period: string) => (
        <td key={period}>
          {householdSize !== '' ? incomeForHouseholdSize[period] : '$XX,XXX'}
        </td>
      ))}
    </tr>
  )
}

export default IncomeRow
