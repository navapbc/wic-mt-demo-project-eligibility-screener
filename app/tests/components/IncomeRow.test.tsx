import { render, screen } from '@testing-library/react'

import IncomeRow from '@components/IncomeRow'

import { testSnapshot } from '../helpers/sharedTests'

const testProps = {
  periods: ['weekly', 'monthly', 'annual'],
  householdSize: '1',
  incomeForHouseholdSize: {
    weekly: '$100',
    monthly: '$200',
    annual: '$300',
  },
}

it('should match snapshot when householdSize is not empty string', () => {
  testSnapshot(<IncomeRow {...testProps} />)
})

it('should match snapshot when householdSize is an empty string', () => {
  testSnapshot(<IncomeRow {...testProps} householdSize="" />)
})
