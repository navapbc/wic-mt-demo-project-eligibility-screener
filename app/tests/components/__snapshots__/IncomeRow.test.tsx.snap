// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot when householdSize is an empty string 1`] = `
<tr>
  <td>
    $XX,XXX
  </td>
  <td>
    $XX,XXX
  </td>
  <td>
    $XX,XXX
  </td>
</tr>
`;

exports[`should match snapshot when householdSize is not empty string 1`] = `
<tr>
  <td>
    $100
  </td>
  <td>
    $200
  </td>
  <td>
    $300
  </td>
</tr>
`;
